### PR TITLE
Added NSPasteBoard and NSGraphicsContext classes

### PR DESCRIPTION
**Pull request by ColleagueRiley**

### CODE REVIEW

File: source/general.m [hunk 1]
```diff
@@ -488,7 +492,39 @@ void NSBezierPath_strokeLine(NSPoint from, NSPoint to) {
...
+}
+/* TODO(EimaMei): This might cause a memory leak, should be looked into later. */
+/* */
+NSInteger NSPasteBoard_declareTypes(NSPasteboard* pasteboard, NSPasteboard** newTypes, NSUInteger array_size, void* owner) {
+	NSArray* new_array = convert_C_array_to_NSArray(newTypes, array_size);
+
+	NSInteger res = [pasteboard declareTypes:(new_array) owner:(owner)];
+	[new_array release];
+
+	return res;
```
Comment (EimaMei) [inline source/general.m:529]: This function should be tested if it includes a memory leak. If the `[new_array release]` line does leak memory, then it's best to remove that line and replace `res` with `return [pasteboard declareTypes:(new_array) owner:(owner)];`

**Review by EimaKve — CHANGES_REQUESTED** [review]
Just check if 'NSPasteBoard_declareTypes' leaks memory and fix it if it does. Other than that it's good

File: include/Silicon/silicon.h [hunk 2]
```diff
@@ -17,7 +17,6 @@ freely, subject to the following restrictions:
    misrepresented as being the original software.
 3. This notice may not be removed or altered from any source distribution.
 */
-
```
Comment (EimaMei) [inline include/Silicon/silicon.h]: There should be a space here

File: include/Silicon/silicon.h [hunk 3]
```diff
@@ -271,7 +270,8 @@ CGFloat NSEvent_deltaY(NSEvent* event);
 unsigned short NSEvent_keyCodeForChar(char* keyStr);
 /* */
 NSPoint NSEvent_mouseLocation(NSEvent* event);
-
+/* */
+NSWindow* NSEvent_window(NSEvent* event);
```
Comment (EimaMei) [inline include/Silicon/silicon.h:275]: There should be another space here.

File: source/general.m [hunk 4]
```diff
@@ -397,7 +397,10 @@ unsigned short NSEvent_keyCodeForChar(char* keyStr){
 CGFloat NSEvent_deltaY(NSEvent* event) {
 	return [event deltaY];
 }
-
+/* */
+NSWindow* NSEvent_window(NSEvent* event){
+	return [event window];
+}
```
Comment (EimaMei) [inline source/general.m:404]: Same as before

**Review by EimaKve — CHANGES_REQUESTED** [review]
Add the spaces and it's good to go

**Review by EimaKve — APPROVED** [review]

**Pull request merged**